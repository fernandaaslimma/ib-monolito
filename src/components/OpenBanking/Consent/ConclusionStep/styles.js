import styled from "styled-components";
import { gray200, gray300 } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const ContainerWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 600px;
  margin: auto;
`;

export const ConclusionStepWrapper = styled.div`
  display: flex;
  flex-direction: column;

  justify-content: center;

  padding: ${rem(16)};

  max-width: ${rem(600)};
  margin: 0 auto;
`;

export const IconWrapper = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;

  margin-top: ${rem(50)};
`;

export const SucessTitle = styled.span`
  font-family: Lato;

  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(20)};
  line-height: 125%;
  /* identical to box height, or 30px */

  text-align: center;

  color: ${gray300};

  margin-top: ${rem(18)};
`;

export const SucessMessage = styled.span`
  font-family: Lato;

  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(16)};
  line-height: 129%;

  text-align: center;

  color: ${gray200};

  margin: ${rem(24)} 0 ${rem(90)};
`;
