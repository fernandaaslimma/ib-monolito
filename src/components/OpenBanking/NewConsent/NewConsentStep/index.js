import NewConsentSetp from "./NewConsentStep";

export default NewConsentSetp;
