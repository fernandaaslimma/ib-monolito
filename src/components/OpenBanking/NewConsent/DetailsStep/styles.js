import styled, { css } from "styled-components";
import {
  gray300,
  white,
  neutral200,
  gray200
} from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const SaveAccount = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-radius: ${rem(6)};
  margin: ${rem(20)} ${rem(12)} 0 ${rem(16)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}

  @media (min-width: 991px) {
    margin-right: ${rem(20)};
  }
`;

export const SaveAccountDeadLine = styled.div`
  display: flex;
  align-items: center;
  border-radius: ${rem(6)};
  margin: 0 ${rem(16)} ${rem(16)} ${rem(16)};
  padding: ${rem(15)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}
`;

export const SaveAccountInfo = styled.div`
  display: flex;
  align-items: center;
`;

export const WapperImg = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(36)};
  width: ${rem(36)};
  border-radius: ${rem(6)};
  border: 1px solid #d9e0e4;
`;

export const SaveAccountInfoDeadLine = styled.div`
  display: flex;
  align-items: center;
  margin-left: ${rem(16)};
`;

export const Text = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(16)};
  font-weight: 600;
  font-style: normal;
  color: #2d4758;
  margin-right: ${rem(16)};
`;

export const CardAccounts = styled.div`
  font-family: Lato;
  font-style: normal;
  font-size: ${rem(14)};
  line-height: ${rem(16.8)};
  padding: ${rem(16)};
  margin-bottom: ${rem(16)};
  background: ${white};
  border: 1px solid #eef1f3;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.08);
  border-radius: 6px;

  ${({ padding }) =>
    padding &&
    css`
      padding: ${rem(padding)} ${rem(16)};
    }
  `}
`;

export const LinkDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  font-family: Lato;
  font-weight: 400;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(16.8)};
  margin-top: ${rem(16)};
  margin-bottom: ${rem(6)};

  ${({ flexStart }) => {
    flexStart &&
      css`
        align-items: flex-start;
        margin-bottom: ${rem(16)};
      `;
  }}
`;

export const LinkCitizen = styled.span`
  text-decoration: underline;
  transition: 0.2s color;
  color: ${neutral200};
  cursor: pointer;
`;

export const WrapperSearch = styled.div`
  margin: ${rem(16)} ${rem(16)} ${rem(24)} ${rem(16)};
  background: #f6f9fb;
`;

export const DescriptionNames = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(22, 4)};
  color: #587485;
  margin-bottom: ${rem(40)};
`;

export const DescriptionNewConsent = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(22, 4)};
  color: #587485;
`;

export const ReceivingInstitutionInfo = styled.div`
  display: flex;
  align-items: center;
  margin-top: ${rem(16)};
`;

export const WrapperInstitution = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const ReceivingInstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 125%;
  /* identical to box height, or 17px */

  letter-spacing: 0.452308px;

  color: ${gray300};

  margin-left: ${rem(16)};
`;

export const AllText = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(12.64)};
  font-weight: 500;
  font-style: normal;
  letter-spacing: ${rem(0.2)};
  margin-right: ${rem(16)};
`;

export const AccountBalance = styled.div`
  display: flex;
  background: #eef1f3;
  box-sizing: border-box;
  border-radius: ${rem(4)};
  align-items: center;
  align-content: center;
  border: ${rem(2)} #eef1f3 solid;
  cursor: pointer;

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: pointer;
    `}

  ${({ noSpacebetween }) =>
    !noSpacebetween &&
    css`
      background: none;
      justify-content: space-between;
    `}
`;

export const AccountText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 400;

  font-size: ${remFontSize(15)};
  line-height: ${rem(20)};
  margin: ${rem(14)} 0 ${rem(14)} ${rem(15)};
  padding: 0;
  color: ${gray200};
`;

export const Value = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: ${rem(12)} ${rem(10)} ${rem(11)} 0;
  padding: 0;
`;

export const Wrapper = styled.div``;

export const Separator = styled.div`
  height: ${rem(1)};
  background-color: #d9e0e4;
`;

export const SubTitle = styled.div`
  font-size: ${remFontSize(12)};
  margin: ${rem(16)} 0 ${rem(11.2)} ${rem(3)};
  font-weight: 500;
  font-style: normal;
  line-height: ${rem(14.4)};
`;

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(12)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(15)};
  color: #587485;
  margin-top: ${rem(16)};
`;

export const Message = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(14)};
  line-height: 150%;
`;

export const MessageBold = styled(Message)`
  font-weight: bold;
`;
