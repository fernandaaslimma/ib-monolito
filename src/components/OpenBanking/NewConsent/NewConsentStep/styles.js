import styled, { css } from "styled-components";
import {
  gray200,
  gray300,
  white,
  neutral200
} from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const AccountBalance = styled.div`
  display: flex;
  background: #eef1f3;
  box-sizing: border-box;
  border-radius: ${rem(4)};
  align-items: center;
  align-content: center;
  border: ${rem(2)} #eef1f3 solid;
  margin: 0 0 0 0;

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: pointer;
    `}

  ${({ noSpacebetween }) =>
    !noSpacebetween &&
    css`
      background: none;
      justify-content: space-between;
    `}
`;

export const AccountText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 400;

  font-size: ${remFontSize(15)};
  line-height: ${rem(20)};
  margin: ${rem(14)} 0 ${rem(14)} ${rem(15)};
  padding: 0;
  color: ${gray200};
`;

export const Value = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: ${rem(12)} ${rem(10)} ${rem(11)} 0;
  padding: 0;
`;

export const ValueInstitution = styled.div`
  display: flex;
  align-items: center;
  margin: ${rem(12)} 0 ${rem(11)} ${rem(10)};
  padding: 0;
`;

export const SaveAccount = styled.div`
  display: flex;
  align-items: center;
  border-radius: ${rem(6)};
  margin: 0 ${rem(16)} ${rem(16)} ${rem(16)};
  padding: ${rem(15)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}
`;
export const SaveAccountInfo = styled.div`
  display: flex;
  align-items: center;
  margin-left: ${rem(16)};
`;

export const Text = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(16)};
  font-weight: 600;
  font-style: normal;
  color: #2d4758;
  margin-right: ${rem(16)};
`;

export const SubTitle = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 600;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(17, 5)};
  color: #2d4758;
`;

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(12)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(15)};
  color: #587485;
  margin-top: ${rem(16)};
`;

export const ReceivingInstitutionInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: ${rem(16)};
  cursor: pointer;
`;

export const WapperImg = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(36)};
  width: ${rem(36)};
  border-radius: ${rem(6)};
  border: 1px solid #d9e0e4;
`;

export const WrapperBottomSheet = styled.div``;

export const ReceivingInstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 125%;
  /* identical to box height, or 17px */

  letter-spacing: 0.452308px;

  color: ${gray300};

  margin-left: ${rem(16)};
`;

export const WrapperInstitution = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const CardAccounts = styled.div`
  display: flex;
  font-family: Lato;
  font-style: normal;
  padding: ${rem(16)};
  margin-bottom: ${rem(24)};
  background: ${white};
  border: 1px solid #eef1f3;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.08);
  border-radius: 6px;
  cursor: pointer;

  ${({ padding }) =>
    padding &&
    css`
      padding: ${rem(padding)} ${rem(16)};
    }
  `}
`;

export const CardAccountsNames = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-family: Lato;
  font-style: normal;
  padding: ${rem(16)};
  margin-bottom: ${rem(24)};
  background: ${white};
  border: 1px solid #eef1f3;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.08);
  border-radius: 6px;
  cursor: pointer;

  ${({ padding }) =>
    padding &&
    css`
      padding: ${rem(padding)} ${rem(16)};
    }
  `}
`;

export const WrapperCardAccounts = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;

  ${({ width }) =>
    width &&
    css`
      width: ${width};
    `}

  ${({ height }) =>
    height &&
    css`
      height: ${height};
    `}
`;

export const WrapperAccounts = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: ${rem(24)};
  white-space: nowrap;
`;

export const TitleAccounts = styled.p`
  font-size: ${remFontSize(14)};
  font-weight: 500;
  color: ${gray300};
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const Bold = styled.span`
  font-weight: bold;
`;

export const LinkDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-family: Lato;
  font-weight: 400;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(16.8)};
  cursor: pointer;
`;

export const LinkCitizen = styled.span`
  text-decoration: underline;
  transition: 0.2s color;
  color: ${neutral200};
`;

export const WrapperSearch = styled.div`
  margin: ${rem(16)} ${rem(16)} ${rem(24)} ${rem(16)};
`;

export const DescriptionNames = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(22, 4)};
  color: #587485;
  margin-bottom: ${rem(40)};
`;

export const TitleNames = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 700;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(16, 8)};
  margin-bottom: ${rem(16)};
  color: #27445f;
`;

export const TitleBig = styled.h2`
  font-size: ${remFontSize(16)};
  margin: 0 0 0 0;
  color: ${gray300};
  font-weight: 700;
  line-height: 125%;
  letter-spacing: 0.452308px;
  margin-top: ${rem(40)};
`;

export const AbsoluteWrapperBottomSheet = styled.div`
  overflow: auto;
  position: sticky;
  bottom: 0;
  background: ${white};
  border-top: solid ${rem(1)} #d3dde4;

  @media (max-width: 991px) {
    position: absolute;
    width: 100%;
  }
`;
