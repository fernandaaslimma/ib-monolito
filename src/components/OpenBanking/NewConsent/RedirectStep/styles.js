import styled from "styled-components";
import { gray200, gray300, white } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const ContainerWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 600px;
  margin: auto;
`;

export const InfoCard = styled.div`
  padding: ${rem(16)} ${rem(15)} ${rem(16)} ${rem(15)};
  background: ${white};
  overflow: auto;
  font-family: Lato;
  font-size: ${remFontSize(14)};
  color: ${gray200};
  line-height: ${remFontSize(22)};
  margin-bottom: ${rem(16)};
`;

export const InfoSpace = styled.div`
  padding: ${rem(16)} ${rem(15)} ${rem(16)} ${rem(15)};
  overflow: auto;
  font-family: Lato;
  font-size: ${remFontSize(14)};
  color: ${gray200};
  line-height: ${remFontSize(22)};
  margin-bottom: ${rem(16)};
`;

export const Title = styled.h2`
  font-size: ${remFontSize(18)};
  margin: 0 0 ${rem(15)} 0;
  color: ${gray300};
  font-weight: 700;
  line-height: 125%;
  /* identical to box height, or 20px */

  letter-spacing: 0.452308px;
`;

export const SubTitle = styled.h2`
  font-size: ${remFontSize(16)};
  margin: 0 0 ${rem(15)} 0;
  color: ${gray300};
  font-weight: 700;
  line-height: 125%;
  /* identical to box height, or 20px */

  letter-spacing: 0.452308px;
`;

export const ReceivingInstitutionInfo = styled.div`
  display: flex;
  align-items: center;
  margin: ${rem(16)} 0;
`;
