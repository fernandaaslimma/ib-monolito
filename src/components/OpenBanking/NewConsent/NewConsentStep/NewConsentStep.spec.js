import React from "react";
import { act } from "react-dom/test-utils";
import { shallow } from "enzyme";
import NewConsentStep from "./NewConsentStep";
import { AccountBalance, TitleAccounts, LinkDetails } from "./styles";

let context = {
  state: {
    selectAccountOriginBottomSheet: false,
    selectedInstitutionBottonSheet: false,
    selectSpecificInstitutionNames: false,
    loadingNewConsentSte: false,
    finalInstitutions: [
      {
        OrganisationId: "f7817b3c-c319-575f-8896-727c1b12cfda",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      },
      {
        OrganisationId: "9415f224-9f58-56b5-af4c-085b4438e4eb",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      }
    ],
    selectedObjective: {
      displayName: "",
      finalityId: ""
    },
    selectFinalCNPJ: "",
    selectedInstitution: {
      CustomerFriendlyLogoUri: "url logo",
      CustomerFriendlyName: "name"
    }
  },
  props: {
    institutions: [
      {
        OrganisationId: "f7817b3c-c319-575f-8896-727c1b12cfda",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      },
      {
        OrganisationId: "9415f224-9f58-56b5-af4c-085b4438e4eb",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      },
      {
        OrganisationId: "f7817b3c-c319-575f-8896-727c1b12cfda",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      },
      {
        OrganisationId: "9415f224-9f58-56b5-af4c-085b4438e4eb",
        AuthorisationServerId: "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
        CustomerFriendlyLogoUri:
          "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
        CustomerFriendlyName: "Carteira BB - HM"
      }
    ],
    specificOrganization: [
      {
        CustomerFriendlyName: "Banco do Brasil"
      },
      {
        CustomerFriendlyName: "Investimentos BB - HM"
      },
      {
        CustomerFriendlyName: "Carteira BB - HM"
      }
    ],
    userInfo: {
      tenants: ["Corporation"],
      document: "3657938596",
      tenantsMembers: [{ document: "algum cnpj" }, { document: " outro" }]
    }
  },
  changeState: jest.fn(),
  getInstituion: jest.fn(),
  createShare: jest.fn()
};

const props = {
  stepForward: jest.fn()
};

describe("NewConsentStep", () => {
  let useEffect, setState, useStateSpy;
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };
  beforeEach(() => {
    jest.useFakeTimers();

    setState = jest.fn();
    useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(init => [init, setState]);

    React.useContext = jest.fn(() => context);
    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();
  });
  it.skip("should match snapshot", () => {
    expect(shallow(<NewConsentStep {...props} />)).toMatchSnapshot();
  });

  it("should view institution", () => {
    const component = shallow(<NewConsentStep {...props} />);
    component
      .find(AccountBalance)
      .at(1)
      .simulate("click");
    expect(context.changeState).toHaveBeenCalled();
  });

  it("should view institution info", () => {
    const component = shallow(<NewConsentStep {...props} />);
    act(() => {
      jest.runOnlyPendingTimers();
    });
    component.update();
    component
      .find(LinkDetails)
      .at(0)
      .simulate("click");
    expect(context.getInstituion).toHaveBeenCalled();
    expect(context.changeState).toHaveBeenCalled();
  });

  it("should match snapshot institution info and back to chose intitutions", () => {
    context.state.selectSpecificInstitutionNames = true;
    React.useContext = jest.fn(() => context);
    const component = shallow(<NewConsentStep {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should select institution", () => {
    const component = shallow(<NewConsentStep {...props} />);
    act(() => {
      jest.runOnlyPendingTimers();
    });
    component.update();
    component
      .find(TitleAccounts)
      .at(0)
      .simulate("click");
    expect(context.changeState).toHaveBeenCalled();
  });
  it("should select institution", () => {
    const component = shallow(<NewConsentStep {...props} />);
    act(() => {
      jest.runOnlyPendingTimers();
    });
    component.update();
    component
      .find(TitleAccounts)
      .at(0)
      .simulate("click");
    expect(context.changeState).toHaveBeenCalled();
  });
});
