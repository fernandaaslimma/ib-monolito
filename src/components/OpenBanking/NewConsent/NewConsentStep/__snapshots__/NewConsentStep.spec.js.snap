// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewConsentStep should match snapshot 1`] = `
<styled.div>
  <InfoCard
    bigTitleMargin="0 0 0 0"
    paddingCard="24px 16px 16px 16px"
    title={
      Object {
        "bigTitle": true,
        "tl": undefined,
      }
    }
  />
  <AlertMessage
    dataTest="alertMessage"
    icon="Attention"
    spacing={
      Object {
        "bottom": "m",
        "left": "s",
        "right": "s",
        "top": "xxs",
      }
    }
    type="neutral"
  />
  <InfoCard
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div />
    <styled.div>
      undefined undefined
    </styled.div>
    <styled.div />
    <styled.div
      data-test="cpfOrCnpjLabel"
    >
      3657938596
    </styled.div>
  </InfoCard>
  <InfoCard
    key="1"
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={false}
      onClick={[Function]}
    >
      <styled.span />
    </styled.div>
  </InfoCard>
  <InfoCard
    key="1"
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={true}
      data-test="openSelectCNPJ"
      onClick={[Function]}
    >
      <styled.span />
      <styled.div>
        <Icon
          color="#3976cf"
          cursor={false}
          fixCrossAxisAlign={null}
          height="18"
          spacing={
            Object {
              "bottom": "none",
              "left": "s",
              "right": "none",
              "top": "none",
            }
          }
          type="Arrow"
          width="18"
        />
      </styled.div>
    </styled.div>
    <AnimatedBottonSheet
      head={
        Object {
          "close": false,
          "title": undefined,
        }
      }
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div />
      <styled.div
        checked={true}
        key="0"
      >
        <Checkbox
          checked={false}
          dataTest="checkboxSelectCNPJ"
          margin="0"
          name="switchBox"
          onChange={[Function]}
          type="common"
        />
        <styled.div>
          <styled.span>
            algum cnpj
          </styled.span>
        </styled.div>
      </styled.div>
      <styled.div
        checked={true}
        key="1"
      >
        <Checkbox
          checked={false}
          dataTest="checkboxSelectCNPJ"
          margin="0"
          name="switchBox"
          onChange={[Function]}
          type="common"
        />
        <styled.div>
          <styled.span>
             outro
          </styled.span>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          <Button
            dataTest="saveCNPJButton"
            disabled={false}
            onClick={[Function]}
            spacing={
              Object {
                "bottom": "s",
                "left": "xxs",
                "right": "s",
                "top": "s",
              }
            }
            type="primary"
          />
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </InfoCard>
  <InfoCard
    title={
      Object {
        "sl": undefined,
        "subTitle": true,
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={true}
      data-test="selectBank"
      noSpacebetween={true}
      onClick={[Function]}
    >
      <styled.div>
        <Icon
          color="#3976cf"
          cursor={false}
          fixCrossAxisAlign={null}
          height="18"
          spacing={
            Object {
              "bottom": "none",
              "left": "s",
              "right": "none",
              "top": "none",
            }
          }
          type="Search"
          width="18"
        />
      </styled.div>
      <styled.span />
    </styled.div>
    <AnimatedBottonSheet
      fullHeight={true}
      head={
        Object {
          "close": true,
          "title": "undefined",
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <SearchBar
          dataTest="Favored"
          keysNameToMach={
            Array [
              "CustomerFriendlyName",
            ]
          }
          list={
            Array [
              Object {
                "AuthorisationServerId": "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
                "CustomerFriendlyLogoUri": "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
                "CustomerFriendlyName": "Carteira BB - HM",
                "OrganisationId": "f7817b3c-c319-575f-8896-727c1b12cfda",
              },
              Object {
                "AuthorisationServerId": "8da36d54-f4ea-4814-ac9a-f08dd2e1f03e",
                "CustomerFriendlyLogoUri": "https://investimentos.bb.com.br/recursos/images/Investimentos_logo.svg",
                "CustomerFriendlyName": "Carteira BB - HM",
                "OrganisationId": "9415f224-9f58-56b5-af4c-085b4438e4eb",
              },
            ]
          }
          maxLength={40}
          placeholder=""
          type="text"
        >
          <styled.div
            key="0"
          >
            <styled.div
              data-test="cardBank_Carteira BB - HM"
              onClick={[Function]}
            >
              <styled.div>
                <Icon
                  color="#2D4758"
                  cursor={false}
                  fixCrossAxisAlign={null}
                  height={20}
                  type="Bank"
                  width={20}
                />
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.p
                onClick={[Function]}
              >
                <styled.span>
                  Carteira BB - HM
                </styled.span>
              </styled.p>
              <styled.div
                height="28px"
                onClick={[Function]}
                width="100%"
              />
              <styled.div
                onClick={[Function]}
              >
                <styled.span />
              </styled.div>
            </styled.div>
            <styled.div
              onClick={[Function]}
              width="100%"
            />
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.div
              data-test="cardBank_Carteira BB - HM"
              onClick={[Function]}
            >
              <styled.div>
                <Icon
                  color="#2D4758"
                  cursor={false}
                  fixCrossAxisAlign={null}
                  height={20}
                  type="Bank"
                  width={20}
                />
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.p
                onClick={[Function]}
              >
                <styled.span>
                  Carteira BB - HM
                </styled.span>
              </styled.p>
              <styled.div
                height="28px"
                onClick={[Function]}
                width="100%"
              />
              <styled.div
                onClick={[Function]}
              >
                <styled.span />
              </styled.div>
            </styled.div>
            <styled.div
              onClick={[Function]}
              width="100%"
            />
          </styled.div>
        </SearchBar>
      </styled.div>
    </AnimatedBottonSheet>
  </InfoCard>
  <styled.div>
    <styled.div>
      <Button
        dataTest="continueNewConsentButton"
        disabled={true}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
        type="primary"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`NewConsentStep should match snapshot institution info and back to chose intitutions 1`] = `
<styled.div>
  <InfoCard
    bigTitleMargin="0 0 0 0"
    paddingCard="24px 16px 16px 16px"
    title={
      Object {
        "bigTitle": true,
        "tl": undefined,
      }
    }
  />
  <AlertMessage
    dataTest="alertMessage"
    icon="Attention"
    spacing={
      Object {
        "bottom": "m",
        "left": "s",
        "right": "s",
        "top": "xxs",
      }
    }
    type="neutral"
  />
  <InfoCard
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div />
    <styled.div>
      undefined undefined
    </styled.div>
    <styled.div />
    <styled.div
      data-test="cpfOrCnpjLabel"
    >
      3657938596
    </styled.div>
  </InfoCard>
  <InfoCard
    key="1"
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={false}
      onClick={[Function]}
    >
      <styled.span />
    </styled.div>
  </InfoCard>
  <InfoCard
    key="1"
    title={
      Object {
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={true}
      data-test="openSelectCNPJ"
      onClick={[Function]}
    >
      <styled.span />
      <styled.div>
        <Icon
          color="#3976cf"
          cursor={false}
          fixCrossAxisAlign={null}
          height="18"
          spacing={
            Object {
              "bottom": "none",
              "left": "s",
              "right": "none",
              "top": "none",
            }
          }
          type="Arrow"
          width="18"
        />
      </styled.div>
    </styled.div>
    <AnimatedBottonSheet
      head={
        Object {
          "close": false,
          "title": undefined,
        }
      }
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div />
      <styled.div
        checked={true}
        key="0"
      >
        <Checkbox
          checked={false}
          dataTest="checkboxSelectCNPJ"
          margin="0"
          name="switchBox"
          onChange={[Function]}
          type="common"
        />
        <styled.div>
          <styled.span>
            algum cnpj
          </styled.span>
        </styled.div>
      </styled.div>
      <styled.div
        checked={true}
        key="1"
      >
        <Checkbox
          checked={false}
          dataTest="checkboxSelectCNPJ"
          margin="0"
          name="switchBox"
          onChange={[Function]}
          type="common"
        />
        <styled.div>
          <styled.span>
             outro
          </styled.span>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          <Button
            dataTest="saveCNPJButton"
            disabled={false}
            onClick={[Function]}
            spacing={
              Object {
                "bottom": "s",
                "left": "xxs",
                "right": "s",
                "top": "s",
              }
            }
            type="primary"
          />
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </InfoCard>
  <InfoCard
    title={
      Object {
        "sl": undefined,
        "subTitle": true,
        "tl": undefined,
      }
    }
  >
    <styled.div
      cursor={true}
      data-test="selectBank"
      noSpacebetween={true}
      onClick={[Function]}
    >
      <styled.div>
        <Icon
          color="#3976cf"
          cursor={false}
          fixCrossAxisAlign={null}
          height="18"
          spacing={
            Object {
              "bottom": "none",
              "left": "s",
              "right": "none",
              "top": "none",
            }
          }
          type="Search"
          width="18"
        />
      </styled.div>
      <styled.span />
    </styled.div>
    <AnimatedBottonSheet
      fullHeight={true}
      head={
        Object {
          "close": false,
          "title": "undefined",
        }
      }
      isOpen={false}
      onClickInBack={[Function]}
      velocity={0.3}
    >
      <styled.div>
        <styled.div />
        <styled.div>
          3 undefined Banco do Brasil.
        </styled.div>
        <styled.div>
          <styled.p
            key="0"
          >
            Banco do Brasil
          </styled.p>
          <styled.p
            key="1"
          >
            Investimentos BB - HM
          </styled.p>
          <styled.p
            key="2"
          >
            Carteira BB - HM
          </styled.p>
        </styled.div>
        <styled.div
          onClick={[Function]}
        >
          <styled.span />
        </styled.div>
      </styled.div>
      <styled.div />
      <styled.div>
        <styled.div>
          <Button
            dataTest="SaveAccountButton"
            disabled={false}
            onClick={[Function]}
            spacing={
              Object {
                "bottom": "s",
                "left": "xxs",
                "right": "s",
                "top": "s",
              }
            }
            type="outline"
          />
        </styled.div>
      </styled.div>
    </AnimatedBottonSheet>
  </InfoCard>
  <styled.div>
    <styled.div>
      <Button
        dataTest="continueNewConsentButton"
        disabled={true}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
        type="primary"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
